test/puppetlabs/puppetdb/generative/submit_command.clj

```diff
@@ -0,0 +1,151 @@
+(ns puppetlabs.puppetdb.generative.submit-command
+  (:require [puppetlabs.puppetdb.command :as cmd]
+            [clojure.test :as t]
+            [clojure.test.check.clojure-test :as tc]
+            [clojure.test.check.generators :as gen]
+            [clojure.test.check.properties :as prop]
+            [puppetlabs.kitchensink.core :as ks]
+            [puppetlabs.puppetdb.testutils.db :refer [*db* with-test-db]]
+            [puppetlabs.puppetdb.cheshire :as json]
+            [puppetlabs.puppetdb.command :as command]
+            [clj-time.core :as time :refer [now]]
+            [puppetlabs.puppetdb.query-eng :as qeng]
+            [puppetlabs.puppetdb.generative.generators :as pdb-gen]
+            [puppetlabs.puppetdb.examples.reports :as example-reports]))
+
+(def ten-timestamps #{"2016-12-19T23:30:00.000Z"
+                      "2016-12-19T23:31:00.000Z"
+                      "2016-12-19T23:32:00.000Z"
+                      "2016-12-19T23:33:00.000Z"
+                      "2016-12-19T23:34:00.000Z"
+                      "2016-12-19T23:35:00.000Z"
+                      "2016-12-19T23:36:00.000Z"
+                      "2016-12-19T23:37:00.000Z"
+                      "2016-12-19T23:38:00.000Z"
+                      "2016-12-19T23:39:00.000Z"})
+
+(defmacro with-catalog-overrides [body]
+  `(with-redefs [pdb-gen/certname #(gen/elements ["a.com" "b.com" "c.com"])
```

Comment:
Talked with @mullr some about this. I think it's really important that we have this kind of override capability. I'd like to see it plumbed through as first class. Sounds like having that gets us a little closer to what the spec version will look like as well which is a nice bonus